CMakeLists.txt

```diff
@@ -543,7 +543,7 @@ set(CPACK_RPM_PRE_UNINSTALL_SCRIPT_FILE "${PROJECT_BINARY_DIR}/packaging/debian/
 # disable automatic shared libraries dependency detection
 set(CPACK_RPM_PACKAGE_AUTOREQ 0)
 
-set(HCC_GENERAL_RPM_DEP "findutils, elfutils-libelf, pciutils-libs, file, pth, libunwind")
+set(HCC_GENERAL_RPM_DEP "findutils, elfutils-libelf, pciutils-libs, file, pth, libunwind, libunwind-devel, gcc-c++, libcxx, libcxxabi")
```

Comment:
We should only include package dependencies for running HCC or codes compiled by HCC here.
Do we really need libunwind-devel and gcc-c++ for that?

Reply:
I included libunwind-devel, because the samples had an issue finding -lunwind. As for gcc-c++, the samples had issues finding g++, and couldn't compile. I ran into these issues from clean docker build, and I thought they might be important to include instead of having the user install it themselves.

Reply:
which samples, are they from HIP?
If a sample needs to be compiled with g++ then that's a sample issue.  We shouldn't make it a dependency for hcc.

Reply:
It seems that for all samples, if we don't have gcc-c++ we will have issues that look like this:
`[root@d8b5fe4b857f hipInfo]#` make
/opt/rocm/hip/bin/hipcc hipInfo.cpp -o hipInfo
Can't exec "g++": No such file or directory at /opt/rocm/hip/bin/hipcc line 134.
Use of uninitialized value $GPP_CUR_VER in substitution (s///) at /opt/rocm/hip/bin/hipcc line 136.
Use of uninitialized value $GPP_CUR_VER in split at /opt/rocm/hip/bin/hipcc line 138.
Use of uninitialized value $GPP_CUR_VER in string eq at /opt/rocm/hip/bin/hipcc line 142.

`[root@d8b5fe4b857f hipDispatchLatency]#` make
/opt/rocm/hip/bin/hipcc -O3 hipDispatchLatency.cpp ResultDatabase.cpp -o hipDispatchLatency
Can't exec "g++": No such file or directory at /opt/rocm/hip/bin/hipcc line 134.
Use of uninitialized value $GPP_CUR_VER in substitution (s///) at /opt/rocm/hip/bin/hipcc line 136.
Use of uninitialized value $GPP_CUR_VER in split at /opt/rocm/hip/bin/hipcc line 138.
Use of uninitialized value $GPP_CUR_VER in string eq at /opt/rocm/hip/bin/hipcc line 142.
WARNING: Linking two modules of different data layouts: '/tmp/tmp.wKtz0NEo1u/hipDispatchLatency-d3ce32.kernel.bc' is '' whereas 'llvm-link' is 'e-p:64:64-p1:64:64-p2:64:64-p3:32:32-p4:32:32-p5:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-v2048:2048-n32:64-A5'

Reply:
It will still make an executable, but running it will result in this:
`[root@d8b5fe4b857f hipDispatchLatency]#` ./hipDispatchLatency

Backtrace:
0x00007f422f32e7a7:     _ZN6Kalmar9HSADevice18IsCompatibleKernelEPvS1_ + 0xe7
0x00007f4231df34a9:     _ZN6Kalmar5CLAMP22DetermineAndGetProgramEPNS_11KalmarQueueEPmPPv + 0x369
0x00007f4231df3941:     _ZN6Kalmar15KalmarBootstrapC2Ev + 0x1a1
0x00007f4231df3779:     __hcc_shared_library_init + 0x29
0x00007f4232087553:     _dl_init_internal + 0x113
0x00007f42320791aa:     check_one_fd.part.0 + 0xaf

### HCC STATUS_CHECK Error: HSA_STATUS_ERROR_INVALID_ISA_NAME (0x1017) at file:/home/jenkins/git/compute_1.6/external/hcc-tot/lib/hsa/mcwamp_hsa.cpp line:2199
Aborted (core dumped)

Reply:
Regarding the "Can't exec g++" warning, that's something in HIP and therefore, the package dependency on g++ should be added to HIP instead of HCC.  Could you please submit a PR to it?

Regarding the invalid ISA name error, could you check whether HIP sets the correct GPU target for your machine?

Reply:
@mangupta Can you help us put the gcc-c++ package into HIP dependencies? I looked into HIP, but I'm not sure how you build packages (its different from HCC which uses CPACK). Could you add gcc-c++ or point me to the correct location?